tests/integration/test_export_and_heudiconv.py

```diff
@@ -159,3 +161,49 @@ def test_heudiconv():
 
     # cleanup output -- for debugging comment this out
     shutil.rmtree(bids_root_dir, ignore_errors=True)
+
+
+def test_unauthorized_user_exception_handling():
+    """Test unauthorized user HTTPError response"""
+    host = "https://xnat.bnc.brown.edu"
+    session = "XNAT_E00114"
+    url = f"{host}/data/experiments/{session}"
+
+    user = "bad_user"
+    password = "bad_password"
+
+    connection = requests.Session()
+    connection.verify = True
+    connection.auth = (user, password)
+
+    try:
+        get(
+            connection,
+            url,
+            params={"format": "json", "handler": "values", "columns": "project,subject_ID"},
+        )
+    except (requests.HTTPError) as e:
+        assert e.response.status_code == 401
```

Comment:
These would make awesome unit tests instead of end-to-end tests. @mirestrepo mentioned how these won't work in CI. Based on my understanding, these tests are specifically trying to test what happens on a 401 or a 404. Have you worked with mocks before? You can create a fake server and force it to respond with whatever status code you need to test your code.

Here's an example library that helps with mocking external server communication. https://httpretty.readthedocs.io/en/latest/

Reply:
We're getting an internal runner soon and we're planning to move them back to unit testing then!